File: bindings/python/examples/stream_events.py
"""Start a run, then watch its durable event deltas as they commit.

kx serve --journal /tmp/kx.db --content /tmp/kx-blobs --dev-allow-local
python examples/stream_events.py
"""

from __future__ import annotations

from kortecx import KxClient


def main() -> None:
    with KxClient("http://127.0.0.1:50151") as kx:
        run = kx.invoke("kx/recipes/echo", {"topic": "watch me"})  # no wait → a handle
        print("watching run", run.instance_id)
        # One snapshot (since the start, to the current journal boundary):
        for delta in run.events(since=0, follow=False):
            print(f"  seq {delta.seq:>4}  {delta.kind:<13} {delta.mote_id or delta.target_mote_id}")
        # For a live tail until Ctrl-C, pass follow=True.


if __name__ == "__main__":
    main()

File: bindings/typescript/examples/stream_events.ts
/**
 * Stream a run's events with a native async iterator.
 *
 * Start a run WITHOUT waiting (get a `Run` handle), then consume its event deltas
 * as they land. `follow: false` reads one snapshot and stops; `follow: true` keeps
 * the live tail open (resuming transparently on a catch-up drop).
 *
 *   npx tsx examples/stream_events.ts
 */

import { KxClient, type Run } from "@kortecx/sdk";

async function main(): Promise<void> {
  const kx = new KxClient("http://127.0.0.1:50151");
  try {
    const run = (await kx.invoke("kx/recipes/echo", { topic: "watch me" })) as Run;
    console.log("instance_id:", run.instanceId);
    for await (const delta of run.events({ since: 0n, follow: false })) {
      console.log(`seq=${delta.seq} ${delta.kind} ${delta.moteId ?? delta.targetMoteId ?? ""}`);
    }
  } finally {
    kx.close();
  }
}

main().catch((err) => {
  console.error(String(err));
  process.exitCode = 1;
});
